src/plugins/workspace/server/integration_tests/duplicate.test.ts

```diff
@@ -0,0 +1,318 @@
+  setSavedObjects: jest.fn(),
+};
+
+export const createExportableType = (name: string): exportMock.SavedObjectsType => {
+  return {
+    name,
+    hidden: false,
+    namespaceType: 'single',
+    mappings: {
+      properties: {},
+    },
+    management: {
+      importableAndExportable: true,
+    },
+  };
+};
+
+describe(`duplicate saved objects among workspaces`, () => {
+  let server: SetupServerReturn['server'];
+  let httpSetup: SetupServerReturn['httpSetup'];
+  let handlerContext: SetupServerReturn['handlerContext'];
+  let savedObjectsClient: ReturnType<typeof savedObjectsClientMock.create>;
+
+  const emptyResponse = { saved_objects: [], total: 0, per_page: 0, page: 0 };
+  const mockIndexPattern = {
+    type: 'index-pattern',
+    id: 'my-pattern',
+    attributes: { title: 'my-pattern-*' },
+    references: [],
+  };
+  const mockVisualization = {
+    type: 'visualization',
+    id: 'my-visualization',
+    attributes: { title: 'Test visualization' },
+    references: [
+      {
+        name: 'ref_0',
+        type: 'index-pattern',
+        id: 'my-pattern',
+      },
+    ],
+  };
+  const mockDashboard = {
+    type: 'dashboard',
+    id: 'my-dashboard',
+    attributes: { title: 'Look at my dashboard' },
+    references: [],
+  };
+
+  beforeEach(async () => {
+    ({ server, httpSetup, handlerContext } = await setupServer());
+    handlerContext.savedObjects.typeRegistry.getImportableAndExportableTypes.mockReturnValue(
+      allowedTypes.map(createExportableType)
+    );
+    handlerContext.savedObjects.typeRegistry.getType.mockImplementation(
+      (type: string) =>
+        // other attributes aren't needed for the purposes of injecting metadata
+        ({ management: { icon: `${type}-icon` } } as any)
+    );
+
+    savedObjectsClient = handlerContext.savedObjects.client;
+    savedObjectsClient.find.mockResolvedValue(emptyResponse);
+    savedObjectsClient.checkConflicts.mockResolvedValue({ errors: [] });
+
+    const router = httpSetup.createRouter('');
+
+    registerDuplicateRoute(router, logger.get(), clientMock, 10000);
+
+    await server.start();
+  });
+
+  afterEach(async () => {
+    await server.stop();
+  });
+
+  it('formats successful response', async () => {
+    clientMock.get.mockResolvedValueOnce({ success: true });
+    exportSavedObjectsToStream.mockResolvedValueOnce(createListStream([]));
+
+    const result = await supertest(httpSetup.server.listener)
+      .post(URL)
+      .send({
+        objects: [
+          {
+            type: 'index-pattern',
+            id: 'my-pattern',
+          },
+          {
+            type: 'dashboard',
+            id: 'my-dashboard',
+          },
+        ],
+        includeReferencesDeep: true,
+        targetWorkspace: 'test_workspace',
+      })
+      .expect(200);
+
+    expect(result.body).toEqual({ success: true, successCount: 0 });
+    expect(savedObjectsClient.bulkCreate).not.toHaveBeenCalled(); // no objects were created
```

Comment:
What is the purpose of this test case? 

Is it testing following scenario:
objects in request body not available in system index (.kibana index). That's why exportSavedObjectsToStream returns empty [].

So for such case user will receive 200 success response with `{ success: true, successCount: 0 }` and no saved object client api will get called. Am I right? If yes, can we change test description to make more clear what it is testing? Thanks!

Reply:
Except above comment, overall looks good to me. Can you please address this comment and rebase your PR?

Reply:
Correct, I've modified the test description, thank you!